src/realm/list.cpp

```diff
@@ -135,16 +135,30 @@ template <class T>
 void Lst<T>::sort(std::vector<size_t>& indices, bool ascending) const
 {
     update_if_needed();
-    auto tree = m_tree.get();
-    if (ascending) {
-        do_sort(indices, size(), [tree](size_t i1, size_t i2) noexcept {
-            return tree->get(i1) < tree->get(i2);
-        });
+
+    if constexpr (std::is_same_v<T, Mixed>) {
+        if (ascending) {
+            do_sort(indices, size(), [this](size_t i1, size_t i2) {
+                return get(i1) < get(i2);
+            });
+        }
+        else {
+            do_sort(indices, size(), [this](size_t i1, size_t i2) {
+                return get(i1) > get(i2);
+            });
+        }
     }
     else {
-        do_sort(indices, size(), [tree](size_t i1, size_t i2) noexcept {
-            return tree->get(i1) > tree->get(i2);
-        });
+        if (ascending) {
```

Comment:
I think it is more efficient to capture the tree pointer as in the original implementation.